src/libs/ReportPrimaryActionUtils.ts

```diff
@@ -0,0 +1,168 @@
+import type {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import type {Policy, Report, Transaction, TransactionViolation} from '@src/types/onyx';
+import {isApprover as isApprovedMember} from './actions/Policy/Member';
+import {getCurrentUserAccountID} from './actions/Report';
+import {hasAccountingConnections, isAutoSyncEnabled, isPrefferedExporter} from './PolicyUtils';
+import {
+    isClosedReport,
+    isCurrentUserSubmitter,
+    isExpenseReport,
+    isHoldCreator,
+    isInvoiceReport,
+    isIOUReport,
+    isOpenReport,
+    isPayer,
+    isProcessingReport,
+    isReportApproved,
+} from './ReportUtils';
+import {getSession} from './SessionUtils';
+import {isDuplicate, isOnHold as isOnHoldTransactionUtils} from './TransactionUtils';
+
+function isSubmitAction(report: Report) {
+    const isExpense = isExpenseReport(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isOpen = isOpenReport(report); 
+    const isManualSubmitEnabled = true; // TODO how to find it?
+
+    return isExpense && isSubmitter && isOpen && isManualSubmitEnabled;
+}
+
+function isApproveAction(report: Report, policy: Policy, reportTransactions: Transaction[]) {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isApprovalEnabled = policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL; // TODO verify
+
+    if (!isExpense || !isApprover || !isApprovalEnabled) {
+        return false;
+    }
+
+    const isOneExpenseReport = isExpenseReport(report) && reportTransactions.length === 1;
+    const isOnHold = reportTransactions.some(isOnHoldTransactionUtils);
+    const isProcessing = isProcessingReport(report);
+    const isOneExpenseReportOnHold = isOneExpenseReport && isOnHold;
+
+    if (isProcessing || isOneExpenseReportOnHold) {
+        return true;
+    }
+
+    return false;
+}
+
+function isPayAction(report: Report, policy: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isReportPayer = isPayer(getSession(), report, false, policy);
+    const isPaymentsEnabled = true; // TODO
```

Comment:
@luacmartins another one. I don't see any auxiliary function that would help

Reply:
@jnowakow you can use this one https://github.com/Expensify/App/blob/6410259e9096d4026f1c3e45b2c0c94b81223d20/src/libs/PolicyUtils.ts#L487